endpoints/openrtb2/race_test.go

```diff
@@ -0,0 +1,106 @@
+package openrtb2
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/exchange"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbsmetrics"
+	"github.com/prebid/prebid-server/stored_requests/backends/empty_fetcher"
+	metrics "github.com/rcrowley/go-metrics"
+)
+
+func TestRequestConcurrently(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(nobidServer))
+	defer server.Close()
+
+	cfg := &config.Configuration{}
+
+	theMetrics := pbsmetrics.NewMetrics(metrics.NewRegistry(), exchange.AdapterList())
```

Comment:
You could also use `pbsmetrics.NewBlankMetrics()` to streamline the metrics code. Without code to actually pipe the metrics to influxdb or somewhere, it shouldn't make a real difference though. And perhaps the load of handling the full metrics code helps bring out the race conditions.